File: Hasil generate/hris-staging_automation.js
import { test, expect } from '@playwright/test';

test('Login and select company PT Risa Staging', async ({ page }) => {
  // Step 1: Navigate to the target URL
  await page.goto('https://hris-staging.kantorku.id/');
  
  // Wait for the page to fully load (network idle)
  await page.waitForLoadState('networkidle');

  // Step 2: Login form - fill in email/username
  // Using placeholder text as a best guess; adjust if actual placeholder differs
  const emailField = page.getByPlaceholder(/email|username|e-mail/i).first();
  await emailField.waitFor({ state: 'visible' });
  await emailField.fill('risa.stagingtest@gmail.com');

  // Step 3: Fill in password
  const passwordField = page.getByPlaceholder(/password|kata sandi/i).first();
  await passwordField.fill('stgtest123!');

  // Step 4: Click login button
  // Trying common roles/text for login button; adjust if needed
  const loginButton = page.getByRole('button', { name: /login|masuk|sign in/i }).first();
  await loginButton.click();

  // Step 5: Wait for navigation to main page / dashboard
  // Assert that we have moved away from the login page
  await expect(page).not.toHaveURL(/login|auth/i, { timeout: 10000 });
  // Optionally wait for the main menu container to appear
  const mainMenu = page.getByRole('navigation').or(page.locator('[data-testid="main-menu"]')).or(page.locator('.main-menu, .sidebar'));
  await mainMenu.waitFor({ state: 'visible', timeout: 15000 });

  // Step 6: Select company "PT Risa Staging"
  // This might be a dropdown, a list item, or a card. Use text matcher.
  // First, check if there is a company selector or if we are already on the selection page.
  // Often after login, HRIS systems show a list of companies to choose from.
  const companyItem = page.getByText('PT Risa Staging', { exact: true })
                         .or(page.getByRole('button', { name: /PT Risa Staging/i }))
                         .or(page.getByRole('option', { name: /PT Risa Staging/i }));
  
  // Wait for the company element to be visible and click it
  await companyItem.first().waitFor({ state: 'visible', timeout: 10000 });
  await companyItem.first().click();

  // Step 7: After selecting company, wait for next page/dashboard to load
  await page.waitForLoadState('networkidle');
  
  // Assert that the user is now on a dashboard or main page (adjust URL as necessary)
  await expect(page).toHaveURL(/dashboard|home|main|company/, { timeout: 10000 });
  
  // Additional check: verify that the selected company name is displayed somewhere on the page
  await expect(page.getByText('PT Risa Staging').first()).toBeVisible({ timeout: 5000 });
});

File: Hasil generate/hris-staging_automation.py
import time
from playwright.sync_api import sync_playwright, expect

def main():
    with sync_playwright() as p:
        # Launch browser (headless=False to see interaction)
        browser = p.chromium.launch(headless=False, slow_mo=500)
        page = browser.new_page()

        try:
            # Navigate to target URL
            page.goto("https://hris-staging.kantorku.id/", wait_until="networkidle")
            page.set_viewport_size({"width": 1366, "height": 768})

            # --- LOGIN ---
            # Wait for email input (expected placeholder or label)
            email_input = page.get_by_placeholder("Email")  # adjust selector if needed
            email_input.wait_for(state="visible", timeout=15000)
            email_input.fill("risa.stagingtest@gmail.com")

            password_input = page.get_by_placeholder("Password")  # adjust selector if needed
            password_input.fill("stgtest123!")

            # Click login button (expected role "button", name "Login")
            login_button = page.get_by_role("button", name="Login")
            login_button.click()

            # Wait for dashboard to load (e.g., presence of sidebar menu)
            page.wait_for_timeout(3000)  # fallback wait
            # Better: wait for a specific element after successful login, e.g., a user avatar or sidebar
            page.wait_for_selector("text=Dashboard", timeout=20000)  # adjust expected text

            # --- NAVIGATE TO KPI ---
            # Click on "KPI" menu in sidebar/navbar
            # Use text matcher, could also be role="link" with name "KPI"
            page.get_by_text("KPI").first.click()
            page.wait_for_timeout(2000)

            # --- ADD KPI ---
            # Click "Add KPI" button (expected text)
            page.get_by_role("button", name="Add KPI").first.click()
            # Or use: page.get_by_text("Add KPI").click()

            # Wait for KPI form modal to appear
            kpi_form = page.locator(".modal-content")  # adjust modal container
            kpi_form.wait_for(state="visible", timeout=10000)

            # Fill KPI Name field (placeholder like "Enter KPI Name")
            page.get_by_placeholder("Enter KPI Name").fill("Test KPI from Automation")

            # Select Target Type: "Basic" (assume dropdown)
            # Locate dropdown by label or aria-label or placeholder
            target_type_dropdown = page.get_by_label("Target Type")  # adjust selector
            target_type_dropdown.select_option("Basic")  # the value might be "basic"

            # If there are other required fields (e.g., target value, period), fill them as needed
            # Example: target value input
            # page.get_by_placeholder("Target Value").fill("100")

            # Submit the form (button might be "Save", "Submit", "Create")
            submit_button = page.get_by_role("button", name="Submit")
            submit_button.click()

            # Wait for success notification
            page.wait_for_timeout(2000)
            success_toast = page.locator(".toast-success")  # adjust toast class
            expect(success_toast).to_be_visible()

            print("KPI added successfully!")

        except Exception as e:
            print(f"An error occurred: {e}")
            # Take screenshot for debugging
            page.screenshot(path="debug_screenshot.png")
        finally:
            browser.close()

if __name__ == "__main__":
    main()